### ISSUE TITLE
defaultCompositeDeletePolicy doesn't apply to XR instantiated directly without Claims

### ISSUE THREAD

**Issue by nello1992** [crossplane/crossplane#5301]
### What happened?
Hi folks! I'm having problems using the defaultCompositeDeletePolicy property applied to an XR instantiated directly without claims.
The official doc reports that this property works both with claims and XRs https://docs.crossplane.io/latest/concepts/composite-resource-definitions/#defaultcompositedeletepolicy, but it doesn't apply to XR with direct instantiation.

My setup involves gitops approach with fluxcd, which applies crossplane resources into the cluster, including the XR. It is, indeed, essential that all resources are deleted respecting the dependencies (set with Usages) and that each XR gets deleted ONLY when all of its resources are deleted first.

How to apply "defaultCompositeDeletePolicy: Foreground" to an XR directly instantiated without using Claims? I know it is possible with "cascade" option as reported on k8s official doc (https://kubernetes.io/docs/tasks/administer-cluster/use-cascading-deletion/#use-foreground-cascading-deletion), but in this case it is not a a viable approach given the gitops adoption.

### What environment did it happen in?
Crossplane version: 1.15.0

**Comment by bobh66** [comment]
Hi @nello1992 - I can see where the documentation might be confusing.  The `defaultCompositeDeletePolicy` applies only to Composites that have been created via Claims.  The Claim controller uses this attribute to determine which propagation policy to use when deleting the associated Composite.

If you are creating Composites directly then you must delete the Composite using the `foreground` propagation policy as described in the foreground cascading deletion documentation.  In this case it would be fluxcd that needs to add the foreground propagation policy to the delete command.

I'll open a PR to clarify the documentation.

**Comment by bobh66** [comment]
@nello1992 - I am updating the documentation to read as follows:

The `defaultCompositeDeletePolicy` defines the deletion policy for composite
resources that are created by claims. The policy is used by the claim controller to determine
which propagation policy to use when it deletes the associated composite.
This policy does not apply to standalone composites that are created without claims.

Using a `defaultCompositeDeletePolicy: Background` policy causes the claim controller to
delete the composite resource and return immediately, relying on Kubernetes to delete the remaining
child objects, like managed resources, nested composites, secrets, etc.

Using `defaultCompositeDeletePolicy: Foreground` causes the claim controller to delete the
associated composite with the propagation policy `foreground` and wait for all child
resources to be deleted before returning.  This causes Kubernetes to use foreground cascading
deletion which ensures that all child resources are deleted before the parent resource is
deleted. Kubernetes deletes all the dependent objects before deleting the composite resource.
The deletion of the claim will suspend until the composite and all of it's child resources
have been deleted.

Does that make more sense?

**Comment by bobh66** [comment]
I realized that there is more documentation missing so I'm going to rework this topic.

**Comment by bobh66** [comment]
I'm not familiar with Flux but ArgoCD allows you to specify the propagation policy to be used when deleting resources: https://argo-cd.readthedocs.io/en/stable/user-guide/sync-options/#resources-prune-deletion-propagation-policy

**Comment by nello1992** [comment]
Hi @bobh66 , thank you for clarifying this interesting point. 
It makes more sense to me now, thanks!
Well, I've devled deep into the flux doc, but it appears it doesn't support this behaviour, unlike argo. So I think the only viable option is to leverage the CI for this purpose.
Based on the implementation of defaultCompositeDeletePolicy mechanism and crossplane architecture, do you think it'd be possibile to add this kind of support in XRs when instantiating them directly? I think that this could be beneficial in these kind of setups and would be independent of gitops tools and strategies adopted.

**Comment by bobh66** [comment]
@nello1992 - I don't know that there is anything we can do from the Crossplane side.  We discussed this briefly when we implemented support for foreground cascading deletion, and in order to override the propagation policy provided by the user (or defaulted by the user) Crossplane would need to take over the responsibility of deleting the entire tree of resources, which we leave to Kuberneters now.  This option was not chosen because it's more work for Crossplane to do what Kubernetes is already doing, and I don't know that we want to override the user's choice of propagation policy.

I think two options are either to have fluxcd deploy claims with `defaultCompositeDeletePolicy: Foreground` or to ask flux to implement support for specifying the propagation policy.
